packages/components/order-tracking/src/order-tracking.controller.js

```diff
@@ -0,0 +1,147 @@
+import angular from 'angular';
+import flatten from 'lodash/flatten';
+import map from 'lodash/map';
+import reverse from 'lodash/reverse';
+
+const ORDER_DEFAULT_HISTORY_LENGTH = 3;
+const SECONDS = 1000;
+const ORDER_DETAILS_POLLING_INTERVAL = 60 * SECONDS;
+const ORDER_FOLLOW_UP_POLLING_INTERVAL = 60 * SECONDS;
+
+export default class OrderTrackingController {
+  /* @ngInject */
+  constructor($log, $q, $timeout, OvhApiMeOrder) {
+    this.$log = $log;
+    this.$q = $q;
+    this.$timeout = $timeout;
+    this.OvhApiMeOrder = OvhApiMeOrder;
+    this.maxHistoryLength = ORDER_DEFAULT_HISTORY_LENGTH;
+  }
+
+  $onInit() {
+    return this.refresh();
+  }
+
+  refresh() {
+    this.order = null;
+    this.orderDetails = null;
+    this.orderFollowUp = null;
+    this.orderHistory = null;
+    this.polling = {
+      orderDetails: null,
+      orderFollowUp: null,
+    };
+    this.error = null;
+    this.isLoading = true;
+
+    this.OvhApiMeOrder.v6().resetAllCache();
+    this.cancelOrderDetailsPolling();
+    this.cancelOrderFollowUpPolling();
+
+    return this.OvhApiMeOrder.v6().get({
+      orderId: this.orderId,
+    }).$promise.then(order => this.getOrderStatus(order).then(({ status }) => {
+      const orderWithStatus = order;
+      orderWithStatus.status = status;
+      return orderWithStatus;
+    })).then((order) => {
+      this.order = order;
+      this.pollOrderDetails(0);
+      this.pollOrderFollowUp(0);
+    }).catch((error) => {
+      this.error = {
+        message: (error.data || { message: error.statusText }).message,
+      };
+      if (angular.isFunction(error.headers)) {
```

Comment:
Weird how you mix `angular` and `lodash`, any reason not to `_.isFunction` (without `_.` of course 😅 ) ?

Reply:
i use angular's helpers when possible and lodash otherwise